File: seeker/snippet/Post.java
//date: 2023-11-30T17:08:19Z
//url: https://api.github.com/gists/42cf15f7e82f395a729602c02f69b521
//owner: https://api.github.com/users/chaeeun-Han

package com.hamahama.pupmory.domain.memorial;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@EntityListeners(AuditingEntityListener.class)
@Builder
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "post_id")
    private Long id;

    @Column(nullable = false)
    private String userUid;

    @Column(columnDefinition = "TEXT")
    private String image;

    @Column(nullable = false)
    private String title;

    private String date;

    private String place;

    private String content;

    private String hashtag;

    @Column(nullable = false)
    private boolean isPrivate;

    @CreatedDate
    private LocalDateTime createdAt;
}

File: seeker/snippet/post.py
#date: 2025-09-01T17:09:08Z
#url: https://api.github.com/gists/bccc52439cea57e1a4369a8c1b4bbc04
#owner: https://api.github.com/users/ubiquitousfire

class Post:
    def __init__(self, post_id, title, subtitle, body):
        self.id = post_id
        self.title = title
        self.subtitle = subtitle
        self.body = body